homeassistant/components/simplisafe/camera.py

```diff
@@ -0,0 +1,243 @@
+"""Support for SimpliSafe cameras."""
+from __future__ import annotations
+
+from datetime import datetime
+import os
+from typing import Any
+
+from simplipy.device import DeviceTypes, DeviceV3
+from simplipy.device.camera import CameraTypes
+from simplipy.errors import SimplipyError
+from simplipy.system.v3 import SystemV3
+from simplipy.websocket import EVENT_CAMERA_MOTION_DETECTED, WebsocketEvent
+import voluptuous as vol
+
+from homeassistant.components.camera import Camera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant, ServiceCall, callback
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.template import Template
+
+from . import SimpliSafe, SimpliSafeEntity
+from .const import DOMAIN, LOGGER
+
+DEFAULT_IMAGE_WIDTH = 720
+
+# The Outdoor Camera has a custom service for saving the motion clip.
+ATTR_FILENAME = "filename"
+ATTR_WIDTH = "width"
+
+SERVICE_OC_IMAGE = "capture_motion_image"
+SERVICE_OC_CLIP = "capture_motion_clip"
+
+SERVICE_OC_IMAGE_SCHEMA = cv.make_entity_service_schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_FILENAME): cv.template,
+        vol.Optional(ATTR_WIDTH, default=720): vol.Coerce(int),
+    }
+)
+
+SERVICE_OC_CLIP_SCHEMA = cv.make_entity_service_schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_FILENAME): cv.template,
+    }
+)
+
+SERVICES = (
+    SERVICE_OC_IMAGE,
+    SERVICE_OC_CLIP,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up SimpliSafe cameras based on a config entry."""
+    simplisafe = hass.data[DOMAIN][entry.entry_id]
+
+    cameras: list[SimplisafeOutdoorCamera] = []
+
+    for system in simplisafe.systems.values():
+        if system.version == 2:
+            LOGGER.info("Skipping camera setup for V2 system: %s", system.system_id)
+            continue
+
+        for camera in system.cameras.values():
+            if camera.camera_type == CameraTypes.OUTDOOR_CAMERA:
+                cameras.append(
```

Comment:
This can be simplified via a comprehension:

```python
cameras = [
    SimpliSafeOutdoorCamera(hass, simplisafe, system, camera)
    for camera in system.cameras.values()
    if camera.camera_type == CameraTypes.OUTDOOR_CAMERA
]
```